otherlibs/dune-rpc/dune_rpc.mli

```diff
@@ -384,4 +384,65 @@ module V1 : sig
         string -> [ `Unix_socket | `Normal_file | `Other ] Fiber.t
     end) : S with type 'a fiber := 'a Fiber.t
   end
+
+  module Registry : sig
+    module File : sig
+      type t =
+        { path : string
+        ; contents : string
+        }
+    end
+
+    module Dune : sig
+      type t
+
+      val where : t -> Where.t
+
+      val root : t -> string
+
+      val of_file : File.t -> t
+    end
+
+    module Config : sig
+      type t
+
+      val create : Xdg.t -> t
+
+      val watch_dir : t -> string
+    end
+
+    type t
+
+    val current : t -> Dune.t list
+
+    module Poll (Fiber : sig
+      type 'a t
+
+      val return : 'a -> 'a t
+
+      val parallel_map : 'a list -> f:('a -> 'b t) -> 'b list t
+
+      module O : sig
+        val ( let* ) : 'a t -> ('a -> 'b t) -> 'b t
+
+        val ( let+ ) : 'a t -> ('a -> 'b) -> 'b t
+      end
+    end) (IO : sig
+      type error
+
+      val scandir : string -> (string list, error) result Fiber.t
+
+      val stat : string -> ([ `Mtime of float ], error) result Fiber.t
```

Comment:
I couldn't resist making the polling a little more efficient by checking mtime. Since files are only added/removed from the registry watch dir, we can avoid reading every file in the dir if the mtime is unmodified.